src/main/java/com/mineboundteam/minebound/item/MyrialSword.java

```diff
@@ -0,0 +1,36 @@
+package com.mineboundteam.minebound.item;
+
+import com.mineboundteam.minebound.entity.MyrialSwordEntity;
+import net.minecraft.world.InteractionHand;
+import net.minecraft.world.InteractionResultHolder;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.SwordItem;
+import net.minecraft.world.item.Tier;
+import net.minecraft.world.level.Level;
+import org.jetbrains.annotations.NotNull;
+
+public class MyrialSword extends SwordItem {
+    MyrialSwordEntity myrialSwordEntity;
+    public int state = 0;
+
+    public MyrialSword(Tier tier, int attackDamageModifier, float attackSpeedModifier, Properties properties) {
+        super(tier, attackDamageModifier, attackSpeedModifier, properties);
+    }
+
+    @Override
+    public @NotNull InteractionResultHolder<ItemStack> use(@NotNull Level level, @NotNull Player player, @NotNull InteractionHand interactionHand) {
```

Comment:
Use will only occur while the item is in the player's hand; we might need to have an invisible texture/model and have an nbt variant we switch to when the sword is thrown if we're going to do it this way. Either that, or another dummy item with no texture to remain in the player's inventory and connect them to the item.